### DISCUSSION TITLE
Automatic changes in configuration?

### DISCUSSION THREAD

**Discussion by doterik** [valinet/ExplorerPatcher#938] [Q&A] [answered]
_Nothing important_ but today, after the upgrade, I got back the Windows 10 start menu. I did not really choose that but used Windows 11, which I don’t mind. It’s the taskbar that is completely useless in Windows 11, which is why your program is so great.

Now the start menu is easy to change but it makes me wonder if more changes have been made in my configuration, ones that are not so clear and may not be discovered until a while, if at all.

It’s not easy to keep up with all the new features that are added... 😉

Although now that I was reminded of the old menu, maybe I should keep it, I had almost forgotten what it was like.

- But the question was whether automatic changes are made in my configuration?

Keep up the good work! 👍🏼

**Comment by Poopooracoocoo** [comment, marked as answer]
I'm guessing you've opted into prereleases? I only opted into them after he added support for the Windows 10 start menu, and I had to enable that myself in Explorer Patcher's properties. I haven't encountered any changes made at all on my system. 

Note that I haven't looked at the code and I'm just a user too, and am just providing my own experience.

**Reply by doterik** [reply]
You're right, I'm using a prerelease.

It doesn't really matter that my preset configuration changed, at least not in this easy-to-detect case.

I am most curious about whether or when values in the configuration change.

So far, I have actively refrained from the Windows 10 menu.

Thanks for your reply.

**Reply by valinet** [reply]
Hi

The newest update did not change any setting. It just uses the `Start_ShowClassicMode` setting from the registry; you probably (surely) had that set to 1 and the program read that and now actually **respects** that. Since 22000.65, Windows ignores that and always presents the Windows 11 menu.

So no, EP did not change any setting automatically. It was there from some time ago when you probably have tinkered with it and EP actually now listens to that and did what it says: enabled the old Windows 10 menu ;)

Thanks

**Reply by doterik** [reply]
Aha, that may very well have been the case, I have for sure tried the Windows 10 start menu before but have actually gotten used to the Windows 11 menu. I don't use it very often so two pages of easily accessible icons are enough for me. However, I was now reminded of tiles on Windows 10 that I missed in the beginning. Mostly a pair of weather tiles but there is now, at least one, on the taskbar. 😉

Thanks for the probable explanation, I'm glad there was one... 👍🏼

You are of course welcome to change settings in the future as long as one is informed and can go through and change at will (and not have to check after every update, that is).

/* For information:
Since 2-3 updates I'm now stuck with this:
![image](https://user-images.githubusercontent.com/4037602/155892657-1259fa21-8803-4532-afa7-3fe4ceec6191.png)
I have tried to empty the cache several times, but that doesn't help.
However, the error message at shutdown, which has been there for a while, now seems to be gone. */

**Reply by valinet** [reply]
> You are of course welcome to change settings in the future

As a principle, I don't like changing random settings without a clear reason and without letting everybody know. EP ships with sane defaults and doesn't change things on its own (there may be bugs, but those should be addressed).

> Since 2-3 updates I'm now stuck with this:

Can you resize or move the widget (drag from its edges or move its titlebar)? Does it happen on [22000.493.42.18](https://github.com/valinet/ExplorerPatcher/releases/tag/22000.493.42.18_a69dffb) as well?

**Reply by doterik** [reply]
I didn't think that you are changing default values without telling, that's why I got confused.

I can resize and move but not scroll.
If I Google something I get disconnected from the weather.

It works with .18.

###### /* I don't know why but I got .19 twice and then .20. That twice thing has happened before. Maybe some delta change. */

**Reply by valinet** [reply]
> /* I don't know why but I got .19 twice and then .20. That twice thing has happened before. Maybe some delta change. */

It happens when I push commits without bumping the version. Nothing to worry about, its normal, not every push is worthy of a version bump, but since you opted into pre-releases, you get everything as soon as it is published.

> It works with .18.

Okay. Do you have `WeatherDevMode` set to 1 in `HKCU\SOFTWARE\ExplorerPatcher`?

**Reply by doterik** [reply]
Yes, `WeatherDevMode` is set to 1.

###### (I didn't worry...)

**Reply by valinet** [reply]
> Yes, `WeatherDevMode` is set to 1.

Then it's normal. Did you enable that on purpose or...? The latest commits introduced a dev mode to the weather widget that aids in debugging (described here: https://github.com/valinet/ExplorerPatcher/wiki/Weather#i-am-facing-a-layout-or-some-other-issue-with-the-content-displayed-by-the-weather-widget-is-there-a-way-to-debug-it). It is designed to help debug the widget in case of various problems, not to be used day to day. One of its features actually is that it shows the whole page. So yeah, it's not a bug, it's a feature. Disable that and the widget will work normally. It's specifically not exposed via the Properties window so that people do not enable accidentally, but only on purpose.

**Reply by doterik** [reply]
I disabled WeatherDevMode and upgraded to .20, now it works again. 😀

You seem to know this program... 

Then I know, thanks!

**Reply by valinet** [reply]
But really, did you enable that on purpose or...? Was this a test or what? 😆

**Reply by doterik** [reply]
I enabled it on purpose. Saw the value in some posts and got interested.

It worked fine, at least for some time without disruption, that's why I saw no connection between the value and the problem, which appeared later. 

Sorry... 😏

**Reply by valinet** [reply]
Yeah, no worries, but my recommendation for the future is not to enable things without at least having a basic/rough understanding of what it does. Whatever, I am glad that it's not actually some new bug 😆

**Reply by doterik** [reply]
👍🏼
